Feat188 patterns weaves (#192)
federated-research/docs: website/pages/federated_research_patterns/computational/traditional.mdx

# Traditional Pattern
**Analytical:** Isolated <br/>
**Data Movement**: Summary <br/>
**Required Trust between TREs**: Low <br/>

> The most common method to undertake federated analytics is for the analyses to be performed in each location individually, summary results are gathered centrally and further meta-style analyses are performed. This pattern requires the lowest trust between TREs, as each result is undertaken independently and a meta or two stage analysis is performed over the results. 

Reviewer: The different meta-analysis methods do map reasonably well onto how we calculate things, but the rationale is not necessarily the same

```suggestion
The most common method of federated analytics carries out a local computation in each location separately to generate summary results which are combined centrally. This pattern requires the lowest trust between TREs, as each result is undertaken independently. 
```

> The limitation of this approach is that data cannot be linked and certain analyses cannot be undertaken via this method. Therefore, the most traditional approach, but the least powerful for a researcher of the three patterns. 

Reviewer: ```suggestion
One limitation of this approach is that data cannot be linked, as no row-level attributes are shared. Certain analyses cannot be undertaken via this method. Though this is the most traditional approach, it is the least powerful of the three patterns for a researcher. 
```

> The analyses that can be run within the boundary of the TRE are almost unlimited and constrained by the compute and permissions from the TRE. The key concept is that manual disclosure control is applied and therefore the results that can then be analysed together are limited. 

Reviewer: Is this necessary with manual disclosure control? The descriptors at the top of the document don't say so

Author: Dont quite follow this one.

Author: @kuraisle could you expand a bit?

Reviewer 2: I think because this is under computational patterns, egress patterns shouldn't come into this bit. We shouldn't be prescribing a particular egress pattern here. If you want to say something along the lines of disclosure control effectively limiting the analyses, I'd probably just change the phrasing slightly, making it a little more open and a little more hypothetical, maybe something like this:

```suggestion
The analyses that can be run within the boundary of the TRE are almost unlimited and constrained by the compute and permissions from the TRE. Disclosure control can be applied which will limit the results that can analysed together. 
```

Either that, or specify an egress pattern for it (it would fit with a manual egress pattern if you're specifically trying to capture the existing federation pattern), and probably move it out of the computational section since it would then encompass more than computational patterns.

Reviewer 2: Thinking about it, maybe the pooled, shared and traditional bits may be better organised in a separate branch of 'common patterns' or 'example patterns' or something like that.

![Alt Text](/assets/gif/Isolated%20Analytics.gif)

## Analyses 
An algorithm to calculate a statistic is *isolated* if the algorithm can be broken into subproblems that can be calculated separately at each TRE. These results can be combined to achieve the same result as calculating them on a single, pooled data set.

> There are often several different algorithms that can be used to calculate the same statistic, but it is the algorithm, not the statistic which is of importance in determining how it can be federated. In considering several common statistics and algorithms in health informatics, we have produced a table which shows statistics that are known to be calculable in a federated manner using Five Safes TES. Note that this summary gives no regard to disclosure risks or checking of outputs, only the suitability of Five Safes TES for an implementation. This table is not comprehensive, and only encompasses some common analyses in health informatics. 

Reviewer: You've talked about "in a federated manner using Five Safes TES" - we were talking on Wednesday about what's possible with 5S-TES. Do you want to be clear that it's about what can be calculated globally from local summaries?

Author: yes

Author: @kuraisle coul you help me out a bit. could you help with this one with a suggestion. Where is that not clear currently?

| Statistic                              | Notes                                                                                |
| -------------------------------------- |  ------------------------------------------------------------------------------------ |
| Minimum/maximum                         |                                                                                      |
| Counts                                 | Includes prevalence, incidence                                                       |
| Contingency tables                     | Includes comorbidity. Can be used for Chi-squared test, Fisher's exact test, etc.    |
| Mean                                   |                                                                                      |
| Variance                               | Can be used to calculate standard deviation                                          |

> | Covariance                             |                                                                                      |

Reviewer: Might be worth mentioning that covariance is a more general example of variance

Author: The table was there before, so is that to add to the table?

| Product-moment correlation coefficient | Also known as Pearson's correlation coefficient. Can be used to calculate Fisher's Z |
| t-tests                                | Including one-sample and two-sample t-tests                                          |
| Hotelling's T2                         |                                                                                      |
| ANOVA                                  |                                                                                      |
| t-digest                               | Used to estimate quartiles (median, interquartile ranges, etc.)                      |

> | Linear regression                     | Can fail on a case by case basis.                                                    |

Reviewer 2: ```suggestion
| Linear regression                     |                                                                                      |
```

> | Linear regression                     | Can fail on a case by case basis.                                                    |

Reviewer: Get @anwarfg to check this - I think he's said it won't fail if you use the right algorithm

Reviewer 2: Yeah, the standard way that I was taught involves inverting a matrix directly, which can fail if the determinant is 0. What you should do in that case is use a stable numerical method like QR decomposition or SVD.

So it won't fail if you use a numerical algorithm instead.

Author: @kuraisle @anwarfg what if anything needs to change?

Author: I think this is resolved.
